### PR TITLE
THRIVE Assessment

### PR DESCRIPTION
**Pull request by awisema**
This is the THRIVE front end, it is not yet integrated into the engagement process.

### CODE REVIEW

File: db/health/migrate/20230525153410_create_thrive_assessments.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class CreateThriveAssessments < ActiveRecord::Migration[6.1]
+  def change
+    create_table :thrive_assessments do |t|
+      t.references :patient
+      t.references :user
```
Comment (eanders) [inline db/health/migrate/20230525153410_create_thrive_assessments.rb]:
```suggestion
      t.references :patient, null: false
      t.references :user, null: false
```

File: drivers/health_thrive_assessment/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+## HealthThriveAssessment README
+
+This README file should be used to explain the functionality of the driver.
```
Comment (eanders) [inline drivers/health_thrive_assessment/README.md]: Replacement for SSM?  Provides the THRIVE assessment for MassHealth CP 2.0.

File: drivers/health_thrive_assessment/app/controllers/health_thrive_assessment/assessments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    before_action :set_patient
+    before_action :set_assessment, only: [:edit, :update, :show, :destroy]
+
+    def new
+      if @patient.thrive_assessments.in_progress.exists?
+        @assessment = @patient.thrive_assessments.in_progress.first
+      else
+        @assessment = @patient.thrive_assessments.create!(user: current_user)
+      end
```
Comment (eanders) [inline drivers/health_thrive_assessment/app/controllers/health_thrive_assessment/assessments_controller.rb]:
```suggestion
      @assessment = if @patient.thrive_assessments.in_progress.exists?
        @patient.thrive_assessments.in_progress.first
      else
        @patient.thrive_assessments.create!(user: current_user)
      end
```

File: drivers/health_thrive_assessment/app/controllers/health_thrive_assessment/assessments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      redirect_to edit_client_health_thrive_assessment_assessment_path(@client, @assessment)
+    end
+
+    def edit
+    end
+
+    def update
+      @assessment.update(assessment_params)
+      redirect_to client_health_careplans_path(@client)
```
Comment (eanders) [inline drivers/health_thrive_assessment/app/controllers/health_thrive_assessment/assessments_controller.rb]: Should this be `resond_with`?  It tends to handle errors more gracefully.  This doesn't seem to care if the update is unable to save...

File: drivers/health_thrive_assessment/app/controllers/health_thrive_assessment/assessments_controller.rb [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    end
+
+    def show
+    end
+
+    def destroy
+      @assessment.destroy
+      redirect_to client_health_careplans_path(@client)
```
Comment (eanders) [inline drivers/health_thrive_assessment/app/controllers/health_thrive_assessment/assessments_controller.rb]: `respond_with`?

File: drivers/health_thrive_assessment/app/controllers/health_thrive_assessment/assessments_controller.rb [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        :help_with_eldercare,
+        :help_with_job_search,
+        :help_with_education,
+        :completed_on,
+      )
+    end
+
+    private def set_patient
+      @client = ::GrdaWarehouse::Hud::Client.find(params[:client_id])
+      @patient = @client.patient
```
Comment (eanders) [inline drivers/health_thrive_assessment/app/controllers/health_thrive_assessment/assessments_controller.rb]: I think there should be some permission check-ing here.  I'm not seeing any of the usual
```
@patient = Health::Patient.accessible_by_user(current_user).find_by(client_id: params[:client_id].to_i)
```
which makes me nervous.

Comment (eanders) [reply drivers/health_thrive_assessment/app/controllers/health_thrive_assessment/assessments_controller.rb]: At the very least, I'd want to see `require_some_patient_access!` at the top.

File: drivers/health_thrive_assessment/app/models/health_thrive_assessment/assessment.rb [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    def help_labels
+      {
+        help_with_food: 'Housing / Shelter',
+        help_with_drug_cost: 'Food',
+        help_with_medical_transportation: 'Paying for Medicine',
+        help_with_utilities: 'Transportation to medical appointments',
+        help_with_childcare: 'Utilities',
+        help_with_eldercare: 'Child care / Daycare',
+        help_with_job_search: 'Care for elder or disabled',
+        help_with_education: 'Job search / training',
```
Comment (eanders) [inline drivers/health_thrive_assessment/app/models/health_thrive_assessment/assessment.rb]: the keys don't align with the values here.  (Off by one?)

File: drivers/health_thrive_assessment/app/views/health_thrive_assessment/assessments/_form.haml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  = f.input :food_worries, as: :select_two, collection: @assessment.food_worries_responses , label: "Within the past 12 months, you worried whether your food would run out before you got money to buy more."
+  = f.input :trouble_drug_cost, as: :select_two, collection: @assessment.yes_no, label: 'Do you have trouble paying for medicines?'
+  = f.input :trouble_medical_transportation, as: :select_two, collection: @assessment.yes_no, label: 'Do you have trouble gettng transportation to medical appointments?'
+  = f.input :trouble_utility_cost, as: :select_two, collection: @assessment.yes_no, label: 'Do you have trouble paying your heating or electricity bill?'
+  = f.input :trouble_caring_for_family, as: :select_two, collection: @assessment.yes_no, label: 'Do you have trouble taking care of a child, family member or friend?'
+  = f.input :unemployed, as: :select_two, collection: @assessment.yes_no, label: 'Are you currently unemployed and looking for a job?'
+  = f.input :interested_in_education, as: :select_two, collection: @assessment.yes_no, label: 'Are you interested in more education?'
```
Comment (eanders) [inline drivers/health_thrive_assessment/app/views/health_thrive_assessment/assessments/_form.haml]: I think these should reference `question_labels` in drivers/health_thrive_assessment/app/models/health_thrive_assessment/assessment.rb. Maybe I'm missing something?

File: drivers/health_thrive_assessment/app/views/health_thrive_assessment/assessments/_form.haml [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  = f.input :help_with_food, as: :pretty_boolean, label: 'Housing / Shelter'
+  = f.input :help_with_drug_cost, as: :pretty_boolean, label: 'Food'
+  = f.input :help_with_medical_transportation, as: :pretty_boolean, label: 'Paying for Medicine'
+  = f.input :help_with_utilities, as: :pretty_boolean, label: 'Transportation to medical appointments'
+  = f.input :help_with_childcare, as: :pretty_boolean, label: 'Utilities'
+  = f.input :help_with_eldercare, as: :pretty_boolean, label: 'Child care / Daycare'
+  = f.input :help_with_job_search, as: :pretty_boolean, label: 'Care for elder or disabled'
+  = f.input :help_with_education, as: :pretty_boolean, label: 'Job search / training'
+
+= f.input :decline_to_answer, as: :pretty_boolean, label: 'I do not want to answer these questions'
```
Comment (eanders) [inline drivers/health_thrive_assessment/app/views/health_thrive_assessment/assessments/_form.haml]: These should reference `help_labels` in `drivers/health_thrive_assessment/app/models/health_thrive_assessment/assessment.rb`

File: drivers/health_thrive_assessment/app/views/health_thrive_assessment/assessments/show.haml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        %tbody
+          - @assessment.question_labels.each do |attr, label|
+            - value = @assessment.send(attr)
+            -next if value.nil?
+            %tr
+              %td= label
+              - if value.in?([true, false])
+                %td= yes_no(value)
+              - else
+                %td= value.humanize
```
Comment (eanders) [inline drivers/health_thrive_assessment/app/views/health_thrive_assessment/assessments/show.haml:38]: I might be wrong, but I think you want to look these up in `housing_statuses` or similar.  The humanized version isn't the same as the related value.

File: drivers/health_thrive_assessment/app/controllers/health_thrive_assessment/assessments_controller.rb (hunk 6 above)
Comment (awisema) [reply drivers/health_thrive_assessment/app/controllers/health_thrive_assessment/assessments_controller.rb]: Ooh! Thanks for catching this! This should be extending `IndividualPatientController`

**Pull request merged**